### PR TITLE
Tones down the demand for Ash

### PR DESCRIPTION
**Pull request by CinnamonSnowball**
## What Does This PR Do
Slightly ups the produced volume of Ash from Oil's temperature reaction.
Tweaks Oil's temperature reaction to only boil and cause a fire if the beaker doesn't have enough empty space remaining (requiring as much empty volume as the oil's), as opposed to every time if the volume is >30u.
Tweaks the Charcoal reaction to require more Salt, effectively lowering the amount of Ash needed.

## Why It's Good For The Game
While burning paper is fine and dandy, it's a bit ridiculous that the self-reliant method requires you to make a dozen batches of Ash 15u at a time to even get anywhere. The sheer amount of Ash you need to make a decent amount of Charcoal is also quite silly.
These changes allow you to make more Ash per batch (depending on the size of your beaker) and tones down how much of it you blow through while making a basic medical healing drug, overall making not relying on paper more viable.

## Testing
Spawned in a small/large/bluespace beaker, made sure you can safely burn up to 25/50/150u of Oil if the rest of the beaker is empty. Made sure a fireflash still happens if there's not enough empty space. Made sure fireflash size scales with boiling overflow volume and can reach the same size as previously.

## Changelog
:cl:
tweak: Heated oil burndown produces slightly more Ash.
tweak: Heated oil burndown reaction now requires its own volume in empty space in the beaker to contain the boiling and avoid causing a fire. This means larger beakers can be used to make more Ash per batch.
tweak: Chemical recipe for Charcoal now requires more Salt (and results in more Charcoal).
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/misc_reagents.dm [hunk 1]
```diff
@@ -178,33 +178,33 @@
 	if(exposed_temperature <= T0C + 600)
 		return
 
-	var/violent = volume > 30
-	var/message_type = violent ? "class='boldwarning'" : "class='warning'"
-	holder.my_atom.visible_message("<span [message_type]>The oil [violent ? "boils and burns violently" : "burns"]!</span>")
+	// Let's say burning boiling oil doubles in volume, can our max volume contain that?
+	var/boil_overflow = (holder.total_volume + volume) - holder.maximum_volume
+	var/violent = boil_overflow > 0
```
Comment (lewcc) [inline code/modules/reagents/chemistry/reagents/misc_reagents.dm:183]: I don't think you need this variable anymore, just make the boil_overflow > 0 check in the if statement.

File: code/modules/reagents/chemistry/reagents/misc_reagents.dm [hunk 2]
```diff
@@ -225,7 +225,6 @@
 	name = "Burned Cooking Oil"
 	description = "It's full of char and mixed with so much crud it's probably useless."
 	id = "cooking_oil_inert"
-	reagent_after_burning = null
```
Comment (DGamerL) [inline code/modules/reagents/chemistry/reagents/misc_reagents.dm:228]: Please also put in the PR somewhere that this also makes ash now!

**Review by DGamerL — CHANGES_REQUESTED** [review]
Please see Lewc's review!

File: code/modules/reagents/chemistry/reagents/misc_reagents.dm (hunk 2 above)
Comment (CinnamonSnowball) [reply code/modules/reagents/chemistry/reagents/misc_reagents.dm:228]: It doesn't, its entire proc is overridden to do nothing; just figured I'd yeet this redundant assignment

File: code/modules/reagents/chemistry/reagents/misc_reagents.dm (hunk 1 above)
Comment (CinnamonSnowball) [reply code/modules/reagents/chemistry/reagents/misc_reagents.dm]: Done, apologies for the wait; I completely forgot about this PR, lmao

File: code/modules/reagents/chemistry/reagents/misc_reagents.dm (hunk 2 above)
Comment (DGamerL) [reply code/modules/reagents/chemistry/reagents/misc_reagents.dm:228]: Oh yeah I see it, amazing

**Review by DGamerL — APPROVED** [review]
LGTM

**Review by S34NW — APPROVED** [review]

**Pull request merged**